app/controllers/assignments_controller.rb

```diff
@@ -294,8 +294,12 @@ def download_test_results
   def stop_test
     test_id = params[:test_run_id].to_i
     assignment_id = params[:id]
-    @current_job = AutotestCancelJob.perform_later(assignment_id, [test_id])
-    session[:job_id] = @current_job.job_id
+    if current_role.student?
+      @current_job = AutotestCancelJob.perform_now(assignment_id, [test_id])
```

Comment:
`perform_now` doesn't return anything, so you don't need to assign it to a variable.